src/Products/Comparison/Service/ComparisonServiceImpl.cs

```diff
@@ -92,178 +95,206 @@ public bool CheckFiles(CompareRequest files)
-            try
+            PreviewOptions previewOptions = new PreviewOptions(pageNumber => result)
             {
-                // get/set parameters
-                string documentGuid = postedData.guid;
-                password = (String.IsNullOrEmpty(postedData.password)) ? null : postedData.password;
-                Comparer comparer = new Comparer();
-                List<PageImage> resultImages = comparer.ConvertToImages(documentGuid, password);
+                PreviewFormat = PreviewFormats.PNG,
+                PageNumbers = new[] { pageNumberToRender },
+                ReleasePageStream = UserReleaseStreamMethod
+            };
 
-                foreach (PageImage page in resultImages)
-                {
-                    PageDescriptionEntity loadedPage = new PageDescriptionEntity();
-                    loadDocumentEntity.SetPages(loadedPage);
-                }
-                return loadDocumentEntity;
-            }
-            catch (Exception ex)
-            {
-                // set exception message
-                throw new FileLoadException("Exception occurred while loading document info", ex);
-            }
+            comparer.Source.GeneratePreview(previewOptions);
+
+            return result;
         }
 
+        private static void UserReleaseStreamMethod(int pageNumber, Stream stream)
+        {
+            stream.Close();
+        }
 
+        private static LoadOptions GetLoadOptions(string password)
+        {
+            LoadOptions loadOptions = new LoadOptions
+            {
+                Password = password
+            };
+
+            return loadOptions;
+        }
 
         private CompareResultResponse CompareTwoDocuments(CompareRequest compareRequest)
         {
             // to get correct coordinates we will compare document twice
             // this is a first comparing to get correct coordinates of the insertions and style changes
-            ICompareResult compareResult = CompareFiles(compareRequest);
             string extension = Path.GetExtension(compareRequest.guids[0].GetGuid());
+            string guid = System.Guid.NewGuid().ToString();
+            //save all results in file
+            string resultGuid = Path.Combine(globalConfiguration.Comparison.GetResultDirectory(), guid + extension);
+
+            Comparer compareResult = CompareFiles(compareRequest, resultGuid);
             ChangeInfo[] changes = compareResult.GetChanges();
-            CompareResultResponse compareResultResponse = GetCompareResultResponse(compareResult, changes, extension);
+
+            CompareResultResponse compareResultResponse = GetCompareResultResponse(changes, resultGuid);
             compareResultResponse.SetExtension(extension);
             return compareResultResponse;
         }
 
-        private ICompareResult CompareFiles(CompareRequest compareRequest)
+        private Comparer CompareFiles(CompareRequest compareRequest, string resultGuid)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'CompareFiles' a static method.](https://app.codacy.com/gh/groupdocs-comparison/GroupDocs.Comparison-for-.NET-MVC/pullRequest?prid=5354506)